apps/web/app/[lang]/blog/article/[slug]/page.tsx

```diff
@@ -0,0 +1,109 @@
+import { useBlog } from '@api/blog/useBlog';
+import { Layout } from '@components/core/layout/Layout';
+
+import { ArticleTags } from './components/ArticleTags';
+import { Title } from './components/Title';
+import { Highlights } from './components/Highlights';
+import { Recommended } from './components/Recommended';
+import { ENDPOINTS } from '@api/endpoints.conts';
+import { BlogPost, BlogPostFilterParams } from '@api/blog/types/blog.types';
+import { ArticleProps } from './Article.types';
+
+const DAY_IN_SECONDS  = 86400
+
+const getRecommendedPosts = async function (
+        tagName:string, 
+        getPostListFn: Function, //(params: BlogPostFilterParams) => Promise<BlogPost[]>
+        getTagIDFn:  Function) { //(tagName: string) => Promise<number> ) {
+    
+    const getPostsParams : BlogPostFilterParams = { 
+        per_page: 4, 
+        page: 1, 
+        context: 'view', 
+        order: 'desc', 
+    }
+    
+    if (tagName) {
+        const tagID = await getTagIDFn(tagName)
+        getPostsParams.tags = [tagID]  
+    }
+    
+    const recommendedPosts = await getPostListFn(getPostsParams)
```

Comment:
Please make sure that all the getter functions for the blog return a value in the catch block to handle this case.

You can let this too if you don't use this constant for anything else:
```suggestion
    return await getPostListFn(getPostsParams)
```